Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: leaving lobby when disconnecting

### Description (*)

With this PR, the host now removes disconnecting clients from their lobby and deletes it when they themselves leave. Meanwhile, the clients also leave the lobby when disconnecting, and after each failed reconnection attempt. This PR also fixes an infinite loop issue that appeared in certain circumstances with RateLimitCooldown by using two operation queues. This fix allow us to put the number of reconnection attempts back at 2.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2701](https://jira.unity3d.com/browse/MTT-2701) and works around the issues caused by [MTT-2667](https://jira.unity3d.com/browse/MTT-2667). Also fixes [MTT-2794](https://jira.unity3d.com/browse/MTT-2794)
### Manual testing scenarios

1. Start a game as a host
2. Join with a client
3. Pause the client until the host times it out
4. Unpause the client and see that it is properly reconnected

This also works when pausing the host to cause a timeout, or when quitting the game as the client or host, or disconnecting via the in-game quit button.

Also works when connection gets denied during approval check (i.e. full game or duplicate connection).
### Questions or comments

### Contribution checklist
